File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.py
class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:
        horizontalCuts.append(0)
        horizontalCuts.append(h)
        horizontalCuts.sort()
        verticalCuts.append(0)
        verticalCuts.append(w)
        verticalCuts.sort()
        max_h = 0
        max_w = 0
        for i in range(len(horizontalCuts)-1):
            if (horizontalCuts[i+1] - horizontalCuts[i]) > max_h:
                max_h = (horizontalCuts[i+1] - horizontalCuts[i])
        for i in range(len(verticalCuts)-1):
            if (verticalCuts[i+1] - verticalCuts[i]) > max_w:
                max_w = (verticalCuts[i+1] - verticalCuts[i])
        return (max_h * max_w) % (10**9 + 7)

File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.cpp
class Solution {
public:
    int maxArea(int h, int w, vector<int>& horizontalCuts, vector<int>& verticalCuts) {
        horizontalCuts.push_back(0);
        horizontalCuts.push_back(h);
        sort(horizontalCuts.begin(), horizontalCuts.end());;
        verticalCuts.push_back(0);
        verticalCuts.push_back(w);
        sort(verticalCuts.begin(), verticalCuts.end());
        int max_h = 0;
        int max_w = 0;
        for (int i=0; i<(horizontalCuts.size() - 1); i++) {
            if ((horizontalCuts[i+1] - horizontalCuts[i]) > max_h) {
                max_h = (horizontalCuts[i+1] - horizontalCuts[i]);
            }
        }
        for (int i=0; i<(verticalCuts.size() - 1); i++) {
            if ((verticalCuts[i+1] - verticalCuts[i]) > max_w) {
                max_w = (verticalCuts[i+1] - verticalCuts[i]);
            }
        }
        long long int res = ((long long int)max_h * (long long int)max_w) % ((long long int)pow(10, 9) + 7);
        return (int) res;
    }
};